static/app/components/compactSelect/types.tsx

```diff
@@ -0,0 +1,36 @@
+import {MenuListItemProps} from 'sentry/components/menuListItem';
+
+export interface SelectOption<Value extends React.Key> extends MenuListItemProps {
+  /**
+   * The option's value, must be unique within the selector.
+   */
+  value: Value;
+}
+
+export interface SelectSection<Value extends React.Key> {
+  options: SelectOption<Value>[];
+  /**
+   * When true, all options inside this section will be disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Optional key to identify this section. If not specified, the section's index will
+   * be used.
+   */
+  key?: React.Key;
+  /**
+   * Title to display on top of section.
+   */
+  label?: React.ReactNode;
+}
```

Comment:
There has been some confusion about what props you can add to select sections (I've seen people add a `value` prop to sections, which doesn't do anything), so this should make it a bit clearer.